src/reference/asciidoc/amqp.adoc

```diff
@@ -410,7 +416,65 @@ The Namespace support also provides the `send-connection-factory-selector-expres
 
 Also starting with _version 1.4_, you can configure a routing connection factory in a `SimpleMessageListenerContainer`.
 In that case, the list of queue names is used as the lookup key.
-For example, if you configure the container with `setQueueNames("foo, bar")`, the lookup key will be `"[foo,bar]"` (no spaces).
+For example, if you configure the container with `setQueueNames("foo", "bar")`, the lookup key will be `"[foo,bar]"` (no spaces).
+
+[[queue-affinity]]
+===== Queue Affinity and the LocalizedQueueConnectionFactory
+
+When using HA queues in a cluster, for the best performance, it can be desirable to connect to the physical broker
+where the master queue resides.
+While the `CachingConnectionFactory` can be configured with multiple broker addresses; this is to fail over and the
+client will attempt to connect in order.
+The `LocalizedQueueConnectionFactory` uses the REST API provided by the admin plugin to determine which node the
+queue is mastered.
+It then creates (or retrieves from a cache) a `CachingConnectionFactory` that will connect to just that node.
+If the connection fails, the new master node is determined and the consumer connects to it.
+The `LocalizedQueueConnectionFactory` is configured with a default connection factory, in case the physical location
+of the queue cannot be determined, in which case it will connect as normal to the cluster.
+
+The `LocalizedQueueConnectionFactory` is a `RoutingConnectionFactory` and the `SimpleMessageListenerContainer` uses the
+queue names as the lookup key as discussed in <<routing-connection-factory>> above.
+
+NOTE: For this reason (the use of the queue name for the lookup), the `LocalizedQueueConnectionFactory` can only be
+used if the container is configured to listen to a single queue.
+
+NOTE: The RabbitMQ management plugin must be enabled on each node.
+
+Here is an example configuration, using Spring Boot's RabbitProperties to configure the factories:
+
+[source, xml]
+----
+@Autowired
+private RabbitProperties props;
+
+private final String[] adminUris = { "http://host1:15672", "http://host2:15672" };
+
+private final String[] nodes = { "rabbit@host1", "rabbit@host2" };
+
+@Bean
+public ConnectionFactory defaultConnectionFactory() {
+    CachingConnectionFactory cf = new CachingConnectionFactory();
+    cf.setAddresses(this.props.getAddresses());
+    cf.setUsername(this.props.getUsername());
+    cf.setPassword(this.props.getPassword());
+    cf.setVirtualHost(this.props.getVirtualHost());
+    return cf;
+}
+
+@Bean
+public ConnectionFactory queueAffinityCF(
+        @Qualifier("defaultConnectionFactory") ConnectionFactory defaultCF) throws Exception{
```

Comment:
I don't see any point from where we throw `Exception` here. So, let me remove it on merge!